classes/webservice/WebserviceOutputBuilder.php

```diff
@@ -35,7 +35,14 @@ class WebserviceOutputBuilderCore
     const VIEW_LIST = 1;
     const VIEW_DETAILS = 2;
 
-    protected $wsUrl;
+    /**
+     * Base PrestaShop webservice URL.
+     *
+     * @var string
+     */
+    public static $wsUrl;
+
+    /** @var ApiNode */
     protected $output;
```

Comment:
As you refactored almost everything in this class, I am quite suspicious with this attribute usage.

As the state of this class is implied in a string construction, what would happen If we use the same builder for several strings ? :thinking: 

The usage workflow of this class is quite difficult to understand reading code from nowhere. Some basic test may help a lot to understand basic usage ^^

Reply:
As far as I can see, the $wsUrl is constructed like this: